src/plugins/testsBundle/index.js

```diff
@@ -18,21 +19,34 @@ module.exports = (kibana) => {
         let modules = [];
         let config = kibana.config;
 
-        // add the modules from all of the apps
-        for (let app of apps) {
-          modules = union(modules, app.getModules());
-        }
+        const testGlobs = ['src/ui/public/**/*.js'];
+        const testingPluginId = config.get('testsBundle.pluginId');
+
+        if (testingPluginId) {
+          const plugin = plugins.byId[testingPluginId];
+
+          // add the modules from all of this plugins apps
+          for (let app of plugin.apps) {
+            modules = union(modules, app.getModules());
+          }
 
-        const testGlobs = [
-          'src/ui/public/**/__tests__/**/*.js',
-        ];
+          testGlobs.push(
+            '!src/ui/public/**/__tests__/**/*',
+            `${plugin.publicDir}/**/__tests__/**/*.js`
+          );
+        } else {
 
-        for (const plugin of plugins) {
-          testGlobs.push(`${plugin.publicDir}/**/__tests__/**/*.js`);
+          // add the modules from all of the apps
+          for (let app of apps) {
+            modules = union(modules, app.getModules());
+          }
+
+          for (const plugin of plugins) {
```

Comment:
Ah, my bad, you're correct.  I was thinking about using `const` in a range-based `for` loop, but in that case it's not an issue of using `const` in the loop so much as with mutating the value of the constant variable within the block.